staging/src/k8s.io/apiserver/pkg/util/flowcontrol/default-objects.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	rmv1a1 "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DefaultPriorityLevelConfigurationObjects returns the instances of
+// the ones defined in the KEP, with system-top in position 0 and
+// workload-low in position 1
+func DefaultPriorityLevelConfigurationObjects() []*rmv1a1.PriorityLevelConfiguration {
+	return []*rmv1a1.PriorityLevelConfiguration{
+		pl("system-top",
+			rmv1a1.PriorityLevelConfigurationSpec{Exempt: true},
+		),
+		pl("workoad-low", rmv1a1.PriorityLevelConfigurationSpec{
+			GlobalDefault:            true,
+			AssuredConcurrencyShares: 100,
+			Queues:                   128,
+			HandSize:                 6,
+			QueueLengthLimit:         100}),
+		pl("system-high", rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 100,
+			Queues:                   128,
+			HandSize:                 6,
+			QueueLengthLimit:         10}),
+		pl("system-low", rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 30,
+			Queues:                   1,
+			QueueLengthLimit:         1000}),
+		pl("workload-high", rmv1a1.PriorityLevelConfigurationSpec{
+			AssuredConcurrencyShares: 30,
+			Queues:                   128,
+			HandSize:                 6,
+			QueueLengthLimit:         100}),
+	}
+}
+
+// DefaultFlowSchemaObjects returns instances of the FlowSchema
+// objects shown in the Example Configuration section of the KEP
+func DefaultFlowSchemaObjects() []*rmv1a1.FlowSchema {
```

Comment:
i thought we agreed on ensuring system objects in the post-start-hook, just like how it's done in `scheduling.Priority`

Reply:
We have a harder problem than the scheduler.  The scheduler can read and write API objects before doing its job (i.e., placing pods).  Our filter's job is to be part of handling _every_ request.  Our filter has to start doing its job before it has had a chance to read all its configuration objects and finish effecting all desired updates.  Otherwise we would tie the system up in a circular dependency.  Please review https://github.com/kubernetes/enhancements/pull/1200

Reply:
so there are two defaults 1) during api server start up, before api server has a chance to read the config 2) during cluster start up, and there is no user/operator provided config.

here you're handling 1), we still need 2), but 2) is left as a separate pr, for 2) you still need to implement it in post start hook. do I understand it right?

Reply:
Technically the function that this comment is attached to is not doing either (1) or (2) but is contributing to both.